.github/workflows/docker.yml

```diff
@@ -21,33 +24,249 @@ jobs:
             echo "has-secrets=1" >> "$GITHUB_OUTPUT"
             echo "has secrets!"
           else
-            echo "has-secrets=0" >> "$GITHUB_OUTPUT"
             echo "no secrets!"
           fi
 
-  docker-build:
+
+  build-lean-image:
+    name: ${{ matrix.image.version }} lean image
     needs: config
-    if: needs.config.outputs.has-secrets
-    name: docker-build
     runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        image:
+          - {version: "3.9-slim-bookworm", suffix: ""}
+          - {version: "3.10-slim-bookworm", suffix: "-py310"}
     steps:
-      - name: "Checkout ${{ github.ref }} ( ${{ github.sha }} )"
-        uses: actions/checkout@v3
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          persist-credentials: false
+      - name: Docker meta
+        id: meta
+        uses: docker/metadata-action@v5
+        with:
+          images: ${{ github.repository }}
+          flavor: |
+            latest=false
+            suffix=${{ matrix.image.suffix }}
+          tags: |
+            type=sha,prefix=,format=long
+            type=ref,event=pr
+            type=raw,value=master,enable={{is_default_branch}}
+            type=raw,value=latest,enable={{is_default_branch}}
```

Comment:
I updated the docker build at the same time that @sebastianliebscher was writing his initial PR so he didn't have a chance to capture the changes from [this PR](https://github.com/apache/superset/pull/25322), but @sfirke and I had updated the docker image `latest` tag to point to the latest official release image only and not the latest production master image so that subsequent docker deployments would run from the latest official release instead of master. There's a [script](https://github.com/apache/superset/pull/25322/files#diff-bcd99032795eb96dd42d38ceeeec262f6d4ab3952e3dbfb75905763c116cd0d7R43) that you can use from the PR to get the value for latest, or feel free to update the way that runs, too.